### PR TITLE
Feature/upgrade to colombe 0.4.0

### PR DESCRIPTION
**Pull request by mikonieminen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,39 +225,48 @@ These tests are still somewhat far from good and you need to validate all result
 
 #### Service tests
 
-These tests are somewhat slow and fragile and because of that these are expected to be run manually.
+These tests are somewhat slow and fragile and because of that these are expect to be ran manually. To create files for your credentials, you need `jq` application installed, if you don't have it or you don't want to install it, you can create these files manually. The format is:
```
Comment (aronerben) [inline README.md]: `expected to be run` was correct, I would say

File: README.md [hunk 2]
```diff
@@ -225,39 +225,48 @@ These tests are still somewhat far from good and you need to validate all result
...
 
 ``` shell
-curl -d '{ "requestor": "letters", "version": "dev" }' "https://api.nodemailer.com/user" -X POST -H "Content-Type: application/json" > ethereal_account.json
+curl -s -d '{ "requestor": "letters", "version": "dev" }' "https://api.nodemailer.com/user" -X POST -H "Content-Type: application/json" | jq '{ hostname: .smtp.host, port: .smtp.port, secure: false, username: .user, password: .pass, }'> ethereal_account.json
 ```
 
-Currently using `ethereal.email` service requires non-released version of `colombe` and
-you need to check out the project, commit `edf757c58fce58c170c63e8a92d3bc81fe4d32ff` contains the needed fix. Then the version with the fix needs to be pinned in the build env:
+Then create *mailtrap.io* account and get your API key:
+- [Signup](https://mailtrap.io/register/signup?ref=header)
```
Comment (aronerben) [inline README.md]: Please lowercase for consistency

File: README.md [hunk 3]
```diff
@@ -225,39 +225,48 @@ These tests are still somewhat far from good and you need to validate all result
...
+- [Copy API token from Settings](https://mailtrap.io/settings)
```
Comment (aronerben) [inline README.md]: Please lowercase for consistency

File: dune-project [hunk 4]
```diff
@@ -21,8 +21,9 @@
   (dune (>= 2.3))
   (emile (>= 1.0))
   (mrmime (>= 0.3.1))
```
Comment (aronerben) [inline dune-project:22]: https://github.com/dinosaure/opam-repository/commit/7b29b54bc7b806246eb2ff8b8f5b766f427b68de (at the bottom of the commit body) suggests we upgrade `mrmime` to `0.3.2` too

File: dune-project [hunk 5]
```diff
@@ -21,8 +21,9 @@
...
-  (colombe (>= 0.3.0))
-  (sendmail-lwt (>= 0.3.0))
+  (colombe (>= 0.4.0))
+  (sendmail (>= 0.4.0))
```
Comment (aronerben) [inline dune-project:24]: Why was this added?

**Review by aronerben — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mikonieminen) [reply README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (mikonieminen) [reply README.md]: Actually `expected to be manually run`. Fixed.

File: README.md (hunk 3 above)
Comment (mikonieminen) [reply README.md]: Fixed.

File: dune-project (hunk 4 above)
Comment (mikonieminen) [reply dune-project:22]: Our dependency is really to `0.3.1` or more recent. I've updated the lock file to have `0.3.2`.

File: dune-project (hunk 5 above)
Comment (mikonieminen) [reply dune-project:24]: I've fixed now all dependencies to be correct. Correct dependency is really to have `sendmail`, because this code does not use `sendmail-lwt`. Code should be refactored to remove `sendmail_handler.*` files and `sendmail-lwt` should be used directly.

File: README.md (hunk 1 above)
Comment (aronerben) [reply README.md]: It is still `expect to be manually run`. Should be `expected`.

File: README.md (hunk 2 above)
Comment (aronerben) [reply README.md]: Thanks

File: README.md (hunk 3 above)
Comment (aronerben) [reply README.md]: Thanks

File: dune-project (hunk 4 above)
Comment (aronerben) [reply dune-project:22]: Thanks!

File: dune-project (hunk 5 above)
Comment (aronerben) [reply dune-project:24]: Ah, was it a transitive dependency before?

File: README.md (hunk 1 above)
Comment (mikonieminen) [reply README.md]: I was wondering why I changed those and after reverting to the original, I realized that grammar checker complained me about these, but I didn't check properly why (it was about using passive voice).  I changed it now quite a lot and actually changed most of this part of the README. Hope it is better now.

Comment (aronerben) [reply README.md]: Looks good, thanks!

**Review by aronerben — APPROVED** [review]
